tests/numeric_test.py

```diff
@@ -197,21 +197,397 @@ def testCumProd(self):
+
+    def testArctan(self):
+        na = np.arange(0, 10, dtype="int64")
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.arctan(na), ak.arctan(pda).to_ndarray()))
+
+        na = np.arange(0, 10, dtype="uint64")
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.arctan(na), ak.arctan(pda).to_ndarray()))
+
+        na = np.linspace(0, 10)
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.arctan(na), ak.arctan(pda).to_ndarray()))
+
+        with self.assertRaises(TypeError):
+            ak.arctan([range(0, 10)])
+
+        # Edge case: infinities
+        na = np.array([np.inf, -np.inf])
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.arctan(na), ak.arctan(pda).to_ndarray(), equal_nan=True))
+
+    def testArctan2(self):
+        na1_int = np.arange(10, 0, -1, dtype="int64")
+        pda1_int = ak.array(na1_int)
+        # this also checks that a divide by zero error is properly handled
+        na2_int = np.arange(0, 10, dtype="int64")
+        pda2_int = ak.array(na2_int)
+
+        na1_uint = np.arange(10, 0, -1, dtype="uint64")
+        pda1_uint = ak.array(na1_uint)
+        na2_uint = np.arange(0, 10, dtype="uint64")
+        pda2_uint = ak.array(na2_uint)
+
+        na1_float = np.linspace(0, 1, 10)
+        pda1_float = ak.array(na1_float)
+        na2_float = np.linspace(0, 10, 10)
+        pda2_float = ak.array(na2_float)
+
+        # vector-vector case
+        self.assertTrue(
+            np.allclose(np.arctan2(na1_int, na2_int), ak.arctan2(pda1_int, pda2_int).to_ndarray())
+        )
+        self.assertTrue(
+            np.allclose(np.arctan2(na1_int, na2_uint), ak.arctan2(pda1_int, pda2_uint).to_ndarray())
+        )
+        self.assertTrue(
+            np.allclose(np.arctan2(na1_int, na2_float), ak.arctan2(pda1_int, pda2_float).to_ndarray())
+        )
+
+        self.assertTrue(
+            np.allclose(np.arctan2(na1_uint, na2_int), ak.arctan2(pda1_uint, pda2_int).to_ndarray())
+        )
+        self.assertTrue(
+            np.allclose(np.arctan2(na1_uint, na2_uint), ak.arctan2(pda1_uint, pda2_uint).to_ndarray())
+        )
+        self.assertTrue(
+            np.allclose(np.arctan2(na1_uint, na2_float), ak.arctan2(pda1_uint, pda2_float).to_ndarray())
+        )
+
+        self.assertTrue(
+            np.allclose(np.arctan2(na1_float, na2_int), ak.arctan2(pda1_float, pda2_int).to_ndarray())
+        )
+        self.assertTrue(
+            np.allclose(np.arctan2(na1_float, na2_uint), ak.arctan2(pda1_float, pda2_uint).to_ndarray())
+        )
+        self.assertTrue(
+            np.allclose(
+                np.arctan2(na1_float, na2_float), ak.arctan2(pda1_float, pda2_float).to_ndarray()
+            )
+        )
+
+        with self.assertRaises(TypeError):
+            ak.arctan2([range(0, 10)], [range(0, 10)])
+
+        # vector-scalar case
+        denom = np.array([5]).astype(np.uint)  # work around to get a scalar uint
```

Comment:
just a reminder, we're prob gonna wanna update this to a more representative uint scalar (like `np.array([2**63 + 1]).astype(np.uint)`). This will need to be done as part of #2508